torchvision/prototype/transforms/_geometry.py

```diff
@@ -13,11 +13,25 @@
 from ._utils import query_image, get_image_dimensions, has_any, is_simple_tensor
 
 
-class HorizontalFlip(Transform):
+class RandomHorizontalFlip(Transform):
+    def __init__(self, p: float = 0.5):
+        super().__init__()
+        self.p = p
+
+    def forward(self, *inputs: Any) -> Any:
+        sample = inputs if len(inputs) > 1 else inputs[0]
+        if torch.rand(1) >= self.p:
+            return sample
+
+        return super().forward(sample)
+
     def _transform(self, input: Any, params: Dict[str, Any]) -> Any:
         if isinstance(input, features.Image):
             output = F.horizontal_flip_image_tensor(input)
             return features.Image.new_like(input, output)
+        elif isinstance(input, features.SegmentationMask):
+            output = F.horizontal_flip_image_tensor(input)
```

Comment:
I created the wrapper, which has the following signature:

```python
def horizontal_flip_segmentation_mask(segmentation_mask: torch.Tensor) -> torch.Tensor:
    pass
```

Not sure about `segmenation_mask` type. I put `torch.Tensor` to be consistent with the other existing wrappers, but ... shouldn't be of type `features.SegmentationMask`?

Reply:
> Not sure about `segmenation_mask` type. I put `torch.Tensor` to be consistent with the other existing wrappers, but ... shouldn't be of type `features.SegmentationMask`?

Nope, the way you implemented it is correct. The low level kernels need to be fully JIT scriptable. Unfortunately, the custom tensors we use like `features.SegmentationMask` are not. Thus, we opted to only have "simple" tensors there and perform all the wrapping on the transforms. 

This of course means that the low level kernels might be more verbose, because one needs to pass in meta data manually, which otherwise wouldn't be necessary:

https://github.com/pytorch/vision/blob/27745e5be086ecdf2a8e609d65c13969cb5201f7/torchvision/prototype/transforms/functional/_geometry.py#L18-L20

https://github.com/pytorch/vision/blob/27745e5be086ecdf2a8e609d65c13969cb5201f7/torchvision/prototype/transforms/_geometry.py#L21-L23